src/AasxServerStandardBib/Database.cs

```diff
@@ -0,0 +1,258 @@
+            return getSubmodelCollection().AsQueryable();
+        }
+        public async void UpdateDBSubmodelById(string submodelIdentifier, ISubmodel newSubmodel)
+        {
+            await getSubmodelCollection().ReplaceOneAsync(r => r.Id.Equals(submodelIdentifier), (Submodel)newSubmodel);
+        }
+        public async void DeleteDBSubmodelById(string submodelIdentifier)
+        {
+            await getSubmodelCollection().DeleteOneAsync(a => a.Id.Equals(submodelIdentifier));
+        }
+        #endregion
+
+        #region ConceptDescription
+        public void WriteDBConceptDescription(IConceptDescription conceptDescription)
+        {
+            try
+            {
+                getConceptDescriptionCollection().InsertOne((ConceptDescription)conceptDescription);
+            }
+            catch (MongoWriteException)
+            {
+            }
+        }
+        public IQueryable<ConceptDescription> GetLINQConceptDescription()
+        {
+            return getConceptDescriptionCollection().AsQueryable();
+        }
+        public async void UpdateDBConceptDescriptionById(IConceptDescription newConceptDescription, string cdIdentifier)
+        {
+            await getConceptDescriptionCollection().ReplaceOneAsync(r => r.Id.Equals(cdIdentifier), (ConceptDescription)newConceptDescription);
+        }
+        public async void DeleteDBConceptDescriptionById(string conceptDescription)
+        {
+            await getConceptDescriptionCollection().DeleteOneAsync(a => a.Id.Equals(conceptDescription));
+        }
+        #endregion
+
+        #region Filestream
+        public async void WriteFile(Stream stream, string filename)
+        {
+            if (stream != null)
+            {
+                Console.WriteLine("New File");
+                var id = await _bucket.UploadFromStreamAsync(filename, stream);
+                stream.Close();
+            }else
+            {
+                //throw new ArgumentNullException(nameof(stream));
+            }
+        }
+        public Stream ReadFile(string filename)
+        {
+            return _bucket.OpenDownloadStream(getFileIdFromFilename(filename));
+        }
+
+        public async void DeleteFile(string filename)
+        {
+            ObjectId fileId = getFileIdFromFilename(filename);
+            await _bucket.DeleteAsync(fileId);
+        }
+
+        private ObjectId getFileIdFromFilename(string filename)
+        {
+            var filter = Builders<GridFSFileInfo>.Filter.Eq(x => x.Filename, filename);
+            var sort = Builders<GridFSFileInfo>.Sort.Descending(x => x.UploadDateTime);
+            var options = new GridFSFindOptions
+            {
+                Limit = 1,
+                Sort = sort
+            };
+
+            using (var cursor = _bucket.Find(filter, options))
+            {
+                var fileInfo = cursor.ToList().FirstOrDefault();
+                // fileInfo either has the matching file information or is null
```

Comment:
This could rather be used as a method summary.

``` csharp
/// <summary>
/// Retrieves the ObjectId of the latest file matching the specified filename in the GridFS bucket.
/// </summary>
/// <param name="filename">The filename of the file to retrieve.</param>
/// <returns>The ObjectId of the matching file, or ObjectId.Empty if no matching file is found.</returns>

```